test/e2e_node/eviction_test.go

```diff
@@ -395,6 +395,26 @@ var _ = SIGDescribe("PriorityMemoryEvictionOrdering", framework.WithSlow(), fram
 			{
 				evictionPriority: 1,
 				pod:              getMemhogPod("high-priority-memory-hog-pod", "high-priority-memory-hog", v1.ResourceRequirements{}),
+				prePodCreationModificationFunc: func(pod *v1.Pod) {
+					nodeList, err := f.ClientSet.CoreV1().Nodes().List(context.Background(), metav1.ListOptions{})
+					framework.ExpectNoError(err)
+					gomega.Expect(nodeList.Items).To(gomega.HaveLen(1))
+
+					nodeSwapInfo := nodeList.Items[0].Status.NodeInfo.Swap
+					if nodeSwapInfo == nil || nodeSwapInfo.Capacity == nil || *nodeSwapInfo.Capacity <= 0 {
+						return
+					}
+
+					// Whenever swap is provisioned on the node, the kernel might be able to reclaim much more memory,
+					// hence it is harder to get the node to be memory pressured. This will add another container that allocates
+					// the same amount as the swap capacity to help bring the node to memory pressure.
+					pod.Spec.Containers = append(pod.Spec.Containers, v1.Container{
```

Comment:
thinking aloud: should this be a sidecar? (in the sense of true kube sidecar container)

Reply:
I'm not sure TBH. What we want here is to add some more stress, but things like startup/termination ordering or similar lifecycle characteristics don't really matter or add value AFAICT. IOW, I don't think this additional container needs any special configuration